sycl/include/sycl/ext/intel/esimd/memory.hpp

```diff
@@ -3902,9 +3918,27 @@ slm_gather(simd<uint32_t, N / VS> byte_offsets, simd_mask<N / VS> mask,
                                         detail::lsc_data_size::default_size>(
         byte_offsets, mask, pass_thru);
   } else {
-    using MsgT = detail::__raw_t<T>;
-    return __esimd_slm_gather_ld<MsgT, N, Alignment>(
-        byte_offsets.data(), mask.data(), pass_thru.data());
+    if constexpr (sizeof(T) == 8) {
+      simd<T, N> Res;
+      Res.template bit_cast_view<uint32_t>().template select<N, 2>(0) =
+          __esimd_slm_gather_ld<uint32_t, N, Alignment>(
```

Comment:
__esimd_slm_gather_ld - generates llvm.masked.gather. Thus there is no need in emulation for 8-byte T.